### PR TITLE
Improve formatter contributor docs

### PR DESCRIPTION
**Pull request by konstin**
The docs were out of date, and the new version incorporates some feedback.

I tried to keep the language concise and the information ordered by how early you need it, so people can get the relevant information quickly before jumping into the code.

I did some minor format_dev changes for consistency in the docs.

### CODE REVIEW

File: crates/ruff_python_formatter/README.md [hunk 1]
```diff
@@ -3,10 +3,205 @@
 The goal of our formatter is to be compatible with Black except for rare edge cases (mostly
 involving comment placement).
 
-## Implementing a node
+## Dev tools
 
-Formatting each node follows roughly the same structure. We start with a `Format{{Node}}` struct
-that implements Default (and `AsFormat`/`IntoFormat` impls in `generated.rs`, see orphan rules below).
+**Testing your changes** You can use the `ruff_python_formatter` binary to format individual files
+and show debug info. It's fast to compile because it doesn't depend on `ruff`. The easiest way is to
```
Comment (MichaReiser) [inline crates/ruff_python_formatter/README.md:9]: :rofl:

File: crates/ruff_python_formatter/README.md [hunk 2]
```diff
@@ -3,10 +3,205 @@
...
+def f():  # a
+    pass
+```
+
+</details>
+
+The other option is to use the playground (also check the playground README):
+
+```shell
+cd playground && npm run dev:wasm && npm install && npm run dev
```
Comment (MichaReiser) [inline crates/ruff_python_formatter/README.md:75]:
```suggestion
cd playground && npm install && npm run dev:wasm && npm run dev
```

File: crates/ruff_python_formatter/README.md [hunk 3]
```diff
@@ -3,10 +3,205 @@
...
+# Get the list of projects
+curl https://raw.githubusercontent.com/akx/ruff-usage-aggregate/master/data/known-github-tomls-clean.jsonl > github_search.jsonl
+# Repurpose this script to download the repositories for us
+python scripts/check_ecosystem.py --checkouts target/checkouts --projects github_search.jsonl -v $(which true) $(which true)
+# Check each project for formatter stability
+cargo run --bin ruff_dev -- format-dev --stability-check --error-file target/formatter-ecosystem-errors.txt --multi-project target/checkouts
+```
+
+**Shrinking** To shrink a formatter error from an entire file to a minimal reproducible example,
+you can use `ruff_shrinking`:
```
Comment (MichaReiser) [inline crates/ruff_python_formatter/README.md:135]: I have to give this a try someday

**Review by MichaReiser — APPROVED** [review]

File: crates/ruff_python_formatter/README.md (hunk 2 above)
Comment (konstin) [reply crates/ruff_python_formatter/README.md]: fwiw the playground readme is still the other way round

**Pull request merged**